app/modules/in-game-lists/abilities.ts

```diff
@@ -1,30 +1,81 @@
+export const REQUIRED_ABILITY_CHUNKS_COUNT = 45;
+
+export const PRIMARY_SLOT_ONLY_REQUIRED_ABILITY_CHUNKS_COUNT = 15;
+
+/**
+ * Reference for Ability Chunks of Primary Slot-Only abilities: https://splatoonwiki.org/wiki/Ability_chunk#Splatoon_3
+ */
 export const abilities = [
-  { name: "ISM", type: "STACKABLE" },
-  { name: "ISS", type: "STACKABLE" },
-  { name: "IRU", type: "STACKABLE" },
-  { name: "RSU", type: "STACKABLE" },
-  { name: "SSU", type: "STACKABLE" },
-  { name: "SCU", type: "STACKABLE" },
-  { name: "SS", type: "STACKABLE" },
-  { name: "SPU", type: "STACKABLE" },
-  { name: "QR", type: "STACKABLE" },
-  { name: "QSJ", type: "STACKABLE" },
-  { name: "BRU", type: "STACKABLE" },
-  { name: "RES", type: "STACKABLE" },
-  { name: "SRU", type: "STACKABLE" },
-  { name: "IA", type: "STACKABLE" },
-  { name: "OG", type: "HEAD_MAIN_ONLY" },
-  { name: "LDE", type: "HEAD_MAIN_ONLY" },
-  { name: "T", type: "HEAD_MAIN_ONLY" },
-  { name: "CB", type: "HEAD_MAIN_ONLY" },
-  { name: "NS", type: "CLOTHES_MAIN_ONLY" },
-  { name: "H", type: "CLOTHES_MAIN_ONLY" },
-  { name: "TI", type: "CLOTHES_MAIN_ONLY" },
-  { name: "RP", type: "CLOTHES_MAIN_ONLY" },
-  { name: "AD", type: "CLOTHES_MAIN_ONLY" },
-  { name: "SJ", type: "SHOES_MAIN_ONLY" },
-  { name: "OS", type: "SHOES_MAIN_ONLY" },
-  { name: "DR", type: "SHOES_MAIN_ONLY" },
+  { name: "ISM", type: "STACKABLE", abilityChunkTypesRequired: [] },
```

Comment:
We can do this, the only thing that I'll require in the code is a check to see if that property exists when attempting to use it:

```js
const primarySlotOnlyAbilityRef = abilities.filter(
        (a) => {
          if (Object.prototype.hasOwnProperty.call(a, 'abilityChunkTypesRequired')) {
            return a.name === mainAbility && (a.abilityChunkTypesRequired.length > 0)
          }
          return false;
        }
      );
```
Without this check, we get a `TypeError: Cannot read properties of undefined (reading 'length')` error

Reply:
An alternative I could do to prevent us having to do a key/property check is create a separate data structure that only contains a `mainAbilityName: subAbilityNames[]` map data structure and then do some data structure linkages, but I feel that keeping everything in the `abilities` const is cleaner at the moment